src/dune_rules/lib.ml

```diff
@@ -1135,9 +1152,67 @@ end = struct
     db.resolve name
     >>= function
     | Ignore -> Memo.return Status.Ignore
+    | Deprecated_library_name (_, name') -> find_internal db name'
     | Redirect_in_the_same_db (_, name') -> find_internal db name'
     | Redirect (db', (_, name')) -> find_internal db' name'
     | Found info -> instantiate db name info ~hidden:None
+    | Multiple_results libs ->
+      let* libs =
+        Memo.List.filter_map
+          ~f:(function
```

Comment:
This is roughly the same treatment for each variant as the one in the "non-multiple" case, but we filter out disabled libs in the `Found` branch.